### DISCUSSION TITLE
Future of the azure iot sdk

### DISCUSSION THREAD

**Discussion by TechWatching** [Azure/azure-iot-sdk-csharp#1828] [Q&A]
Is it planned for the Azure iot SDK to be rewritten as the [other Azure SDKs](https://azure.github.io/azure-sdk/) ? All other Azure SDK have been rewritten to be consistent, easily testable, and adopt common patterns. It would be nice to have the same for the azure iot sdk.

**Comment by drwill-ms** [comment]
Yep, we plan to update them to the (what we call) track2 standard. We have to weigh that against the planned new features in IoT Hub, DPS, TSI and other services we support, so we're not able to work on it right now. It is heavily on our minds though.
